### PR TITLE
Feature/directory structure

### PR DESCRIPTION
**Pull request by jcwenger**

### CODE REVIEW

File: openauto/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,117 @@
+add_library(openauto SHARED
+        App.cpp
+        Service/BluetoothService.cpp
```
Comment (stefan-sherwood) [inline openauto/CMakeLists.txt:3]: It's not necessary to explicitly enumerate all of the files.

File: btservice/btservice.cpp [hunk 2]
```diff
@@ -33,22 +33,22 @@ int main(int argc, char* argv[])
     btservice::AndroidBluetoothServer androidBluetoothServer;
     if(!androidBluetoothServer.start(address, portNumber))
     {
-        OPENAUTO_LOG(error) << "[btservice] Server start failed.";
+        OPENAUTO_LOG(error) << "[src] Server start failed.";
```
Comment (stefan-sherwood) [inline btservice/btservice.cpp]: Why this change? The old way seems to make more sense.

File: include/openauto/Service/AndroidAutoEntity.hpp [hunk 3]
```diff
@@ -19,21 +19,17 @@
...
-#include <f1x/aasdk/Channel/AV/VideoServiceChannel.hpp>
-#include <f1x/openauto/autoapp/Configuration/IConfiguration.hpp>
-#include <f1x/openauto/autoapp/Service/IAndroidAutoEntity.hpp>
-#include <f1x/openauto/autoapp/Service/IService.hpp>
-#include <f1x/openauto/autoapp/Service/IPinger.hpp>
+#include "aasdk/Transport/ITransport.hpp"
+#include "aasdk/Channel/Control/IControlServiceChannel.hpp"
+#include "aasdk/Channel/Control/IControlServiceChannelEventHandler.hpp"
+#include "aasdk/Channel/AV/VideoServiceChannel.hpp"
+#include "../Configuration/IConfiguration.hpp"
```
Comment (stefan-sherwood) [inline include/openauto/Service/AndroidAutoEntity.hpp]: `..` is generally bad form. Is there another way?

File: openauto/CMakeLists.txt (hunk 1 above)
Comment (stefan-sherwood) [reply openauto/CMakeLists.txt:3]: For example: https://stackoverflow.com/questions/57355474/how-to-add-library-with-pattern-for-sources

Comment (jcwenger) [reply openauto/CMakeLists.txt:3]: https://cmake.org/cmake/help/v3.14/command/file.html?highlight=file#filesystem

> Note We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild.

File: btservice/btservice.cpp (hunk 2 above)
Comment (jcwenger) [reply btservice/btservice.cpp]: Whoops.  This should not have changed.  Will revert.

File: include/openauto/Service/AndroidAutoEntity.hpp (hunk 3 above)
Comment (jcwenger) [reply include/openauto/Service/AndroidAutoEntity.hpp]: CLion renamed a couple of these and added relative paths.   Missed this one, thanks.  Will re-scrub.

File: openauto/CMakeLists.txt (hunk 1 above)
Comment (stefan-sherwood) [reply openauto/CMakeLists.txt:3]: Fair enough. I wasn't necessarily advocating that particular method. I was just suggesting that having to maintain a list of "all files in a directory" is kind of silly. If there is a way to do it with a regex or just by adding by directory name that reduces dependencies and makes the code easier to read.
